geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/CreateJndiBindingCommand.java

```diff
@@ -165,24 +156,36 @@ else if (alreadyExists)
     if (dsConfigProperties != null && dsConfigProperties.length > 0)
       configuration.setDatasourceConfigurations(Arrays.asList(dsConfigProperties));
 
+    Result result;
+    boolean persisted = false;
     ClusterConfigurationService service = getSharedConfiguration();
 
-    if (service == null) {
-      return ResultBuilder.createUserErrorResult("Cluster Configuration Service is not available. "
-          + "Please connect to a locator with running Cluster Configuration Service.");
+    if (service != null) {
+      boolean alreadyExists = isBindingAlreadyExists(jndiName);
+      if (alreadyExists && ifNotExists)
+        return ResultBuilder.createInfoResult(CliStrings.format(
+            "Skipping creation of Jndi Binding with jndi-name \"{0}\" as it already exists.",
+            jndiName));
+      else if (alreadyExists)
+        return ResultBuilder.createUserErrorResult(
+            CliStrings.format("JNDI Binding with jndi-name \"{0}\" already exists.", jndiName));
+
+      updateXml(configuration);
+      persisted = true;
     }
 
-    updateXml(configuration);
-
     Set<DistributedMember> targetMembers = findMembers(null, null);
     if (targetMembers.size() > 0) {
       List<CliFunctionResult> jndiCreationResult =
           execute(new CreateJndiBindingFunction(), configuration, targetMembers);
-      return buildResult(jndiCreationResult);
+      result = buildResult(jndiCreationResult);
+    } else {
+      result = ResultBuilder.createInfoResult("No members found.");
```

Comment:
In this case, if cc is not available, user will get a message saying "cluster config is not updated", but if cc is available, there is no more message. User might get the impression that nothing happens. Maybe in that case, we should say something like "No member found. Cluster configuration is updated"